### PR TITLE
Cleanup merge.py file

### PR DESCRIPTION
**Pull request by mpozniak95**
## Description
Move code related with perf profiling from merge.py file to perf.py file.

## Related Issue
https://github.com/Granulate/gprofiler/issues/672

### CODE REVIEW

File: gprofiler/merge.py [hunk 1]
```diff
@@ -29,16 +28,9 @@
...
-    re.MULTILINE | re.DOTALL,
-)
-
 # ffffffff81082227 mmput+0x57 ([kernel.kallsyms])
 # 0 [unknown] ([unknown])
 # 7fe48f00faff __poll+0x4f (/lib/x86_64-linux-gnu/libc-2.31.so)
-FRAME_REGEX = re.compile(r"^\s*[0-9a-f]+ (.*?) \(\[?(.*?)\]?\)$")
 
 
 def parse_one_collapsed(collapsed: str, add_comm: Optional[str] = None) -> StackToSampleCount:
```
Comment (mpozniak95) [inline gprofiler/merge.py:36]: Not sure what to do with these three functions. I don't think they belong here. Maybe somewhere in utils?

Comment (mpozniak95) [reply gprofiler/merge.py:36]: I mean: `parse_one_collapsed`, `parse_one_collapsed_file `and `parse_many_collapsed`

Comment (Jongy) [reply gprofiler/merge.py:36]: Yeah, utils. You can add a file `gprofiler/utils/collapsed_format.py` and put them there

File: gprofiler/merge.py [hunk 2]
```diff
@@ -5,168 +5,24 @@
...
 
 logger = get_logger_adapter(__name__)
 
-SAMPLE_REGEX = re.compile(
-    r"\s*(?P<comm>.+?)\s+(?P<pid>[\d-]+)/(?P<tid>[\d-]+)(?:\s+\[(?P<cpu>\d+)])?\s+(?P<time>\d+\.\d+):\s+"
-    r"(?:(?P<freq>\d+)\s+)?(?P<event_family>[\w\-_/]+):(?:(?P<event>[\w-]+):)?(?P<suffix>[^\n]*)(?:\n(?P<stack>.*))?",
-    re.MULTILINE | re.DOTALL,
-)
-
 # ffffffff81082227 mmput+0x57 ([kernel.kallsyms])
```
Comment (Jongy) [inline gprofiler/merge.py:23]: Please move this comment as well, it belongs to FRAME_REGEX

**Review by Jongy — APPROVED** [review]
LGTM

**Pull request merged**